omod/src/main/java/org/openmrs/module/dhisreport/web/controller/MapReportsController.java

```diff
@@ -72,4 +76,59 @@ public void mapReportsSubmit( ModelMap model, @RequestParam( value = "reportDefi
         reportDefinition.setReportingReportId( reportmoduleDefinitionID );
         service.saveReportDefinition( reportDefinition );
     }
+    
+    @RequestMapping( value = "/module/dhisreport/confirmReports", method = RequestMethod.POST )
+    public void confirmReports( ModelMap model, @RequestParam( value = "reporting_report", required = true )
+    int reportIndex, @RequestParam( value = "dhis_report_id", required = true )
+    int dhisReportId )
+    {
+        try
+        {
+            DHIS2ReportingService service = Context.getService( DHIS2ReportingService.class );
+            DataValueTemplate selected = service.getDataValueTemplate( dhisReportId );
+            org.openmrs.module.reporting.report.definition.ReportDefinition rrd = Context.getService(
+                ReportDefinitionService.class ).getDefinitionByUuid( rd.getReportingReportId() );
+            PeriodIndicatorReportDefinition pird = (PeriodIndicatorReportDefinition) rrd;
+            SqlCohortDefinition scd = null;
+            if ( reportIndex > 0 )
+            {
+                CohortIndicatorDataSetDefinition cidsd = pird.getIndicatorDataSetDefinition();
+                scd = (SqlCohortDefinition) cidsd.getColumns().get( reportIndex - 1 ).getIndicator()
+                    .getParameterizable().getCohortDefinition().getParameterizable();
+            }
+            if ( selected.getDefaultreportquery() != null )
+            {
+                if ( reportIndex == 0 )
+                {
+                    selected.setQuery( selected.getDefaultreportquery() );
+                    selected.setDefaultreportquery( null );
+                    selected.setMappeddefinitionlabel( null );
+                }
+                else
+                {
+                    selected.setQuery( scd.getQuery() );
+                }
+            }
+            else
+            {
+                if ( reportIndex == 0 )
+                {
+                    selected.setQuery( selected.getDefaultreportquery() );
+                    selected.setDefaultreportquery( null );
+                    selected.setMappeddefinitionlabel( null );
+                }
+                else
+                {
+                    selected.setDefaultreportquery( selected.getQuery() );
+                    selected.setMappeddefinitionlabel( scd.getName() );
+                    selected.setQuery( scd.getQuery() );
+                }
+            }
+            service.saveDataValueTemplate( selected );
+        }
+        catch ( Exception e )
+        {
+            System.out.println( e.getMessage() );
```

Comment:
Did you see this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout